internal/config/projectconfig/init.go

```diff
@@ -59,3 +44,33 @@ func Init(dir string, projectName string, projectContext *ZeroProjectConfig) {
 		exit.Fatal(fmt.Sprintf("Failed to create config file %s", constants.ZeroProjectYml))
 	}
 }
+
+func GetProjectFileContent(projectConfig ZeroProjectConfig) string {
+	var tpl bytes.Buffer
+	tmpl := template.New("sa")
+	tmpl, err := tmpl.Parse(exampleConfig)
+	if err != nil {
+		exit.Fatal(fmt.Sprintf("Failed to parse the sample Zero module config file %s", constants.ZeroProjectYml))
+	}
+
+	type tempProjectConfig struct {
+		Name                   string
+		ShouldPushRepositories bool
+		Modules                string
+	}
+
+	foo, err := yaml.Marshal(projectConfig.Modules)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Should this fail here? Return this error?

Reply:
@bmonkman this is something I'm not 100% certain about and could use some clarity on, I've refactored it to exit with non-zero if serialization of the modules fails but I'm a bit unsure if I should rather propagate the error to the caller in this case `Init` too handle it there.

Reply:
In most cases you should return the error to the caller to let it decide what to do.

Reply:
Unless something absolutely fatal happened. In this case, seems better to return an error. Easier to test too.